From ca6af270d8a1289abaa90ed3767031d347b0f21a Mon Sep 17 00:00:00 2001
From: Luca Weiss <luca@z3ntu.xyz>
Date: Thu, 11 Apr 2019 14:08:15 +0200
Subject: [PATCH] Migrate from libnm-glib to libnm

libnm-glib is deprecated, not built by all distros anymore and will soon
be removed completely from NetworkManager
---
 CMakeLists.txt                              |  8 +-------
 tests/integration/test-connectivity-api.cpp | 11 ++++++-----
 2 files changed, 7 insertions(+), 12 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 685d4578..031664da 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -59,12 +59,6 @@ pkg_check_modules(
 )
 include_directories(${OFONO_INCLUDE_DIRS})
 
-pkg_check_modules(
-    NETWORKMANAGER REQUIRED
-    NetworkManager
-)
-include_directories(${NETWORKMANAGER_INCLUDE_DIRS})
-
 pkg_check_modules(
   LIBSECRET REQUIRED
   libsecret-1
@@ -77,7 +71,7 @@ include_directories(${Qt5Core_INCLUDE_DIRS})
 find_package(Qt5DBus COMPONENTS Qt5DBusMacros REQUIRED)
 include_directories(${Qt5DBus_INCLUDE_DIRS})
 
-pkg_check_modules(NM REQUIRED NetworkManager REQUIRED)
+pkg_check_modules(NM REQUIRED libnm REQUIRED)
 include_directories(${NM_INCLUDE_DIRS})
 
 pkg_check_modules(URL_DISPATCHER REQUIRED url-dispatcher-1)
diff --git a/tests/integration/test-connectivity-api.cpp b/tests/integration/test-connectivity-api.cpp
index 777f3f2e..70397a3c 100644
--- a/tests/integration/test-connectivity-api.cpp
+++ b/tests/integration/test-connectivity-api.cpp
@@ -20,6 +20,7 @@
 #include <indicator-network-test-base-phone.h>
 #include <dbus-types.h>
 #include <NetworkManagerSettingsInterface.h>
+#include <nm-dbus-interface.h>
 
 #include <QDebug>
 #include <QTestEventLoop>
@@ -383,7 +384,7 @@ TEST_F(TestConnectivityApi, HotspotConfig)
 
     auto& nmSettingsMock = dbusMock.mockInterface(NM_DBUS_SERVICE,
                            NM_DBUS_PATH_SETTINGS,
-                           NM_DBUS_IFACE_SETTINGS,
+                           NM_DBUS_INTERFACE_SETTINGS,
                            QDBusConnection::SystemBus);
     QSignalSpy nmSettingsMockCallSpy(
                            &nmSettingsMock,
@@ -454,7 +455,7 @@ TEST_F(TestConnectivityApi, HotspotConfig)
     // Connect to method calls on the newly added connection
     auto connectionPath = qvariant_cast<QDBusObjectPath>(settingsNewConnectionSpy.first().first());
     auto& connectionSettingsMock = dbusMock.mockInterface(NM_DBUS_SERVICE, connectionPath.path(),
-                           NM_DBUS_IFACE_SETTINGS_CONNECTION,
+                           NM_DBUS_INTERFACE_SETTINGS_CONNECTION,
                            QDBusConnection::SystemBus);
     QSignalSpy connectionSettingsMockCallSpy(
                            &connectionSettingsMock,
@@ -538,7 +539,7 @@ TEST_F(TestConnectivityApi, InsecureHotspotConfig)
 
     auto& nmSettingsMock = dbusMock.mockInterface(NM_DBUS_SERVICE,
                            NM_DBUS_PATH_SETTINGS,
-                           NM_DBUS_IFACE_SETTINGS,
+                           NM_DBUS_INTERFACE_SETTINGS,
                            QDBusConnection::SystemBus);
     QSignalSpy nmSettingsMockCallSpy(
                            &nmSettingsMock,
@@ -603,7 +604,7 @@ TEST_F(TestConnectivityApi, InsecureHotspotConfig)
     // Connect to method calls on the newly added connection
     auto connectionPath = qvariant_cast<QDBusObjectPath>(settingsNewConnectionSpy.first().first());
     auto& connectionSettingsMock = dbusMock.mockInterface(NM_DBUS_SERVICE, connectionPath.path(),
-                           NM_DBUS_IFACE_SETTINGS_CONNECTION,
+                           NM_DBUS_INTERFACE_SETTINGS_CONNECTION,
                            QDBusConnection::SystemBus);
     QSignalSpy connectionSettingsMockCallSpy(
                            &connectionSettingsMock,
@@ -633,7 +634,7 @@ TEST_F(TestConnectivityApi, HotspotModemAvailable)
 
     auto& nmSettingsMock = dbusMock.mockInterface(NM_DBUS_SERVICE,
                            NM_DBUS_PATH_SETTINGS,
-                           NM_DBUS_IFACE_SETTINGS,
+                           NM_DBUS_INTERFACE_SETTINGS,
                            QDBusConnection::SystemBus);
     QSignalSpy nmSettingsMockCallSpy(
                            &nmSettingsMock,
-- 
2.21.0

